core/domain/draft_upgrade_services.py

```diff
@@ -134,89 +134,107 @@ def _convert_html_in_draft_change_list(
             # The change object has the key 'new_value' only if the
             # cmd is 'CMD_EDIT_STATE_PROPERTY' or
             # 'CMD_EDIT_EXPLORATION_PROPERTY'.
-            new_value = change.new_value
-            if change.property_name == exp_domain.STATE_PROPERTY_CONTENT:
-                # Ruling out the possibility of any other type for MyPy
-                # type checking, because here 'new_value' is a value of
-                # attribute that was dynamically defined on change_domain
-                # and that attribute has str type. Thus to convert the
-                # type, we used assert here.
-                assert isinstance(new_value, dict)
-                new_value['html'] = conversion_fn(new_value['html'])
-            elif (change.property_name ==
+            # Here we use cast because we are narrowing down the type from
+            # ExplorationChange to a specific change command.
+            edit_state_property_cmd = cast(
+                exp_domain.EditExplorationStatePropertyCmd,
+                change
+            )
+            new_value = edit_state_property_cmd.new_value
+            if (
+                edit_state_property_cmd.property_name ==
+                exp_domain.STATE_PROPERTY_CONTENT
+            ):
+                # Here we use cast because in this 'if clause' we are
```

Comment:
I think this is not so great -- you'll want subtypes of EditExplorationStatePropertyCmd instead that condition on the property name, and should deal with the typing there, not here. I.e. cast the entire edit_state_property_cmd.

Ditto for the others.

Reply:
New subtypes are defined!